java/rocksjni/portal.h

```diff
@@ -9089,5 +9089,155 @@ class BlockBasedTableOptionsJni
   }
 };
 
+class WideColumnJni {
+ public:
+  static void convertWideColumnsDirect(
+      std::function<void(ROCKSDB_NAMESPACE::Slice&,
+                         ROCKSDB_NAMESPACE::WideColumns&)>
+          op,
+      JNIEnv* env, jobject jKey, jint jKeyOffset, jint jKeyLen,
+      jobjectArray jNames, jintArray jNamesOffset, jintArray jNamesLen,
+      jobjectArray jValues, jintArray jvaluesOffset, jintArray jValuesLen) {
+    auto _key = env->GetDirectBufferAddress(jKey);
+    if (_key == nullptr) {
+      ROCKSDB_NAMESPACE::RocksDBExceptionJni::ThrowNew(
+          env,
+          "Invalid key argument (argument is not a valid direct ByteBuffer)");
+      return;
+    }
+
+    auto key = reinterpret_cast<char*>(_key) + jKeyOffset;
+    ROCKSDB_NAMESPACE::Slice key_slice(key, static_cast<size_t>(jKeyLen));
+
+    const auto columns_len = static_cast<int>(env->GetArrayLength(jNames));
+
+    auto namesOffset = std::make_unique<jint[]>(columns_len);
```

Comment:
make_unique seems an unnecessary overhead here?

Reply:
Yes, that's true, but it allows me to automatically release all the memory allocated here and simplify return statements.